File: 2849-determine-if-a-cell-is-reachable-at-a-given-time/2849-determine-if-a-cell-is-reachable-at-a-given-time.py
class Solution(object):
    def isReachableAtTime(self, sx, sy, fx, fy, t):
        
        if sx == fx and sy == fy and t == 1:
            return False

        farther = None
        x = abs(sx - fx)
        y = abs(sy - fy)
        if x > y:
            farther = x
        else: 
            farther = y

        if farther > t:
            return False
        else: 
            return True

File: 2849-determine-if-a-cell-is-reachable-at-a-given-time/2849-determine-if-a-cell-is-reachable-at-a-given-time.c
bool isReachableAtTime(int sx, int sy, int fx, int fy, int t){
    if (sx == fx && sy == fy && t == 1){
        return false;
    }
    
    int farther = 0;
    int x = abs(sx - fx);
    int y = abs(sy - fy);
    if (x > y){
        farther = x;
    } else {
        farther = y;
    }

    if (farther > t){
        return false;
    } else {
        return true;
    }
}